### PR TITLE
Add ecotone support to `op_alloy_rpc_types::OptimismTransactionReceiptFileds`

### PR DESCRIPTION
**Pull request by emhane**
## Motivation

Closes https://github.com/alloy-rs/op-alloy/issues/57

## Solution

Port fields from depreciated alloy type https://github.com/alloy-rs/alloy/commit/ab462f12ac40fd16bd9882c9451d7d62f7c31d37

## PR Checklist

- [ ] Added Tests
- [x] Added Documentation
- [x] Breaking changes

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
lgtm, you're right this should be simplified once we integrate this with the OpTxReceipt, which has the deposit_ fields already.

but for reth we need this as is rn

File: crates/rpc-types/src/receipt.rs [hunk 1]
```diff
@@ -34,29 +34,57 @@ impl ReceiptResponse for OpTransactionReceipt {
...
+    ///
+    /// Always null prior to the Canyon hardfork.
     #[serde(default, skip_serializing_if = "Option::is_none", with = "alloy_serde::quantity::opt")]
-    pub l1_gas_used: Option<u128>,
+    pub deposit_receipt_version: Option<u64>,
+    /* ---------------------------------------- Ecotone ---------------------------------------- */
+    /// L1 base fee scalar. Applied to base fee to compute weighted gas price multiplier.
+    ///
+    /// Always null prior to the Ecotone hardfork.
+    pub l1_base_fee_scalar: Option<u128>,
```
Comment (mattsse) [inline crates/rpc-types/src/receipt.rs:78]: missing attribute
```suggestion
    #[serde(default, skip_serializing_if = "Option::is_none", with = "alloy_serde::quantity::opt")]
    pub l1_base_fee_scalar: Option<u128>,
```

File: crates/rpc-types/src/receipt.rs [hunk 2]
```diff
@@ -34,29 +34,57 @@ impl ReceiptResponse for OpTransactionReceipt {
...
+    /// L1 blob base fee.
+    ///
+    /// Always null prior to the Ecotone hardfork.
+    #[serde(default, skip_serializing_if = "Option::is_none", with = "alloy_serde::quantity::opt")]
+    pub l1_blob_base_fee: Option<u128>,
+    /// L1 blob base fee scalar. Applied to blob base fee to compute weighted gas price multiplier.
+    ///
+    /// Always null prior to the Ecotone hardfork.
+    pub l1_blob_base_fee_scalar: Option<u128>,
```
Comment (mattsse) [inline crates/rpc-types/src/receipt.rs:87]:
```suggestion
    #[serde(default, skip_serializing_if = "Option::is_none", with = "alloy_serde::quantity::opt")]
    pub l1_blob_base_fee_scalar: Option<u128>,
```

**Review by mattsse — CHANGES_REQUESTED** [review]

**Comment by emhane** [comment]
> lgtm, you're right this should be simplified once we integrate this with the OpTxReceipt, which has the deposit_ fields already.
> 
> but for reth we need this as is rn

correct, since we're still using `alloy_rpc_types_eth::AnyTransactionReceipt` in reth RPC

**Pull request merged**